src/main/java/com/avispl/symphony/dal/communicator/crestron/CrestronXiO.java

```diff
@@ -179,395 +319,505 @@ private void processAvailableDevicesMetadata() throws Exception
-	 * In order to propagate accountId required by Crestron we use "login" property that Symphony is aware of
-	 *
-	 * @return value of Crestron API account ID
-	 */
-	@Override
-	public String getLogin() {
-		return getAccountId();
-	}
-
-	/**
-	 * This method is for compatibility with Symphony communicator infrastructure
-	 * as Symphony is not capable of setting arbitrary properties on the communicator class.
-	 * In order to propagate accountId required by Crestron we use "login" property that Symphony is aware of
-	 *
-	 * @param login value of Crestron API account ID
-	 */
-	@Override
-	public void setLogin(String login) {
-		setAccountId(login);
-	}
-
-	/**
-	 * This method is for compatibility with Symphony communicator infrastructure
-	 * as Symphony is not capable of setting arbitrary properties on the communicator class.
-	 * In order to propagate subscriptionId required by Crestron we use "password" property that Symphony is aware of
-	 *
-	 * @return value of Crestron API subscription ID
-	 */
-	@Override
-	public String getPassword() {
-		return getSubscriptionId();
-	}
-
-	/**
-	 * This method is for compatibility with Symphony communicator infrastructure
-	 * as Symphony is not capable of setting arbitrary properties on the communicator class.
-	 * In order to propagate subscriptionId required by Crestron we use "password" property that Symphony is aware of
-	 *
-	 * @param password value of Crestron API subscription ID
-	 */
-	@Override
-	public void setPassword(String password) {
-		setSubscriptionId(password);
-	}
-
-	/**
-	 * Temporarily used method for testing purposes
-	 */
-	/**
-	public static void main(String[] args) throws Exception {
-		CrestronXiO xio = new CrestronXiO();
-
-		xio.setHost("hqe-api.crestron.io");
-		xio.setProtocol("https");
-		xio.setPort(443);
-		xio.setLogin("cc39be8b-217f-447c-b23c-4be37136283c");
-		xio.setPassword("77f660fb9156494b888365fa559ff20d");
-
-		try {
-			xio.init();
-		} catch (Exception e) {
-			e.printStackTrace();
-		}
-
-		while (true)
-		{
-			List<AggregatedDevice> aggregatedDevices = xio.retrieveMultipleStatistics();
-			System.out.println("*aggregatedDevices = " + aggregatedDevices.toString());
-			Thread.sleep(10*1000);
-		}
-	}
-	 */
-}
+        JsonNode deviceStatistics = fetchDeviceStatistics(deviceId);
+        updateAggregatedDevice(deviceStatistics);
+    }
+
+    /**
+     * Retrieves information about available devices.
+     *
+     * @return {@link JsonNode} instance with information about available devices
+     */
+    private JsonNode fetchAvailableDevices() throws Exception {
+        JsonNode response = doGet("api/v1/device/accountid/" + getAccountId() + "/devices", JsonNode.class);
+        controlLock.lock();
+        try {
+            if (response != null && !response.isNull() && response.size() > 0) {
+                validDeviceMetaDataRetrievalPeriodTimestamp = System.currentTimeMillis() + Math.max(120000, deviceMetaDataInformationRetrievalTimeout);
```

Comment:
Hard to decipher hardcoded values. Move 120000 to constants please and describe it there.